aidial_adapter_bedrock/llm/model/anthropic.py

```diff
@@ -145,4 +164,130 @@ async def _apredict(
             consumer.append_content(content)
         consumer.close_content()
 
-        consumer.add_usage(compute_usage(prompt, completion))
+        consumer.add_usage(self._compute_usage(prompt, completion))
+
+    def _compute_usage(self, prompt: str, completion: str) -> TokenUsage:
+        batch = self.tokenizer.encode_batch([prompt, completion])
+
+        return TokenUsage(
+            prompt_tokens=len(batch[0].ids),
+            completion_tokens=len(batch[1].ids),
+        )
+
+
+class UsageEventHandler(AsyncMessageStream):
```

Comment:
You could declared the members in the class explicitely with the defaults.
And also specify the type of `stop_reason` more precisely.

Pattern match over all cases of `stop_reason` to make the typing tighter in `to_dial_finish_reason` method.

```
prompt_tokens: int = 0
completion_tokens: int = 0
stop_reason: Optional[Literal["end_turn", "max_tokens", "stop_sequence"]] = None
```

Reply:
Done.